src/librustc_typeck/collect.rs

```diff
@@ -1575,14 +1575,10 @@ fn predicates_defined_on<'a, 'tcx>(
     def_id: DefId,
 ) -> ty::GenericPredicates<'tcx> {
     let explicit = tcx.explicit_predicates_of(def_id);
-    let predicates = if tcx.sess.features_untracked().infer_outlives_requirements {
-        [
-            &explicit.predicates[..],
-            &tcx.inferred_outlives_of(def_id)[..],
-        ].concat()
-    } else {
-        explicit.predicates
-    };
+    let predicates = [
+      &explicit.predicates[..],
+      &tcx.inferred_outlives_of(def_id)[..],
```

Comment:
I *think* if we change this to something like:

```rust
let span = tcx.hir.span_if_local(def_id).unwrap();
&tcx.at(span).inferred_outlives_of(def_id)[..]
```

we might fix some of those cycle errors to have "cycle used at" messages again. That said, I don't think this is a big win and we probably shouldn't do it -- in particular, the span will just point at the declaration itself, which is really just kind of confusing.

Reply:
So never mind =)